packages/components/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,164 @@
+      />
+    </StyledCheckboxIndicator>
+  );
+};
+
+export interface CheckboxRootProps extends RadixCheckboxProps {
+  wrapped?: boolean;
+  error?: boolean;
+  CheckboxIcon?: React.FunctionComponent;
+  children: NonNullable<React.ReactNode>;
+}
+
+const getCheckboxStyles = (
+  wrapped: boolean
+): {
+  backgroundColor: SystemProp<keyof Theme["colors"], Theme>;
+  padding: SystemProp<keyof Theme["space"], Theme>;
+} => {
+  if (wrapped) {
+    return {
+      backgroundColor: "colorBackgroundWeak",
+      padding: "space40",
+    };
+  }
+  return {
+    backgroundColor: "colorBackground",
+    padding: "space0",
+  };
+};
+
+const StyledCheckboxRoot = styled.input`
+  width: 16px;
+  height: 16px;
+  border-radius: 4px;
+  margin-right: ${theme.space.space30};
+  padding: 4px 0 0;
+
+  &:focus {
+    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.05), 0 0 0 2px ${
+      theme.colors.colorBackground
+    }, 0 0 0 4px ${theme.colors.colorBackgroundPrimary};
+    outline: none;
+  },
+
+  &[data-state=unchecked] {
+    background-color: ${theme.colors.colorBackground};
+    border: 1px solid ${(props: { error: boolean }) =>
+      props.error
+        ? theme.colors.colorBackgroundDestructive
+        : theme.colors.colorBorder};
+
+    &:hover {
+      background-color: ${theme.colors.colorBackgroundStrong};
+      border: 1px solid ${theme.colors.colorBorder};
+    },
+  },
+  
+  &[data-state=checked], &[data-state=indeterminate] {
+    background-color: ${theme.colors.colorBackgroundPrimary};
+    border: 1px solid ${theme.colors.colorBackgroundPrimary};
+
+    &[data-disabled=true] {
+      background-color: ${theme.colors.colorBackgroundStrong};
+      border: 1px solid ${theme.colors.colorBackgroundStrong};
+    }
+
+    &:hover {
+      background-color: ${theme.colors.colorBackgroundPrimaryStrong};
+      border: 1px solid ${theme.colors.colorBackgroundPrimaryStrong};
+    },
+  },
+  
+  &[data-disabled=true] {
+    background-color: ${theme.colors.colorBackground};
+    border: 1px solid ${theme.colors.colorBackgroundStrong};  
+  
+    &:hover {
+      background-color: ${theme.colors.colorBackgroundStrong};
+      border: 1px solid ${theme.colors.colorBackgroundStrong};
+    },
+  },
+`;
+
+export const Root = React.forwardRef<HTMLButtonElement, CheckboxRootProps>(
+  (
+    {
+      wrapped = false,
+      error = false,
+      CheckboxIcon = CheckIndicator,
+      children,
+      ...props
+    },
+    ref
+  ) => {
+    return (
+      <Box.div
+        alignItems="center"
+        borderRadius="borderRadius20"
+        display="flex"
+        {...getCheckboxStyles(wrapped)}
+      >
+        <StyledCheckboxRoot as={RadixRoot} error={error} ref={ref} {...props}>
+          <CheckboxIcon />
+        </StyledCheckboxRoot>
+        <Box.div
+          color={error ? "colorTextError" : "colorTextStronger"}
+          fontSize="sm"
+          marginBottom="space10"
+        >
+          {children}
+        </Box.div>
```

Comment:
Is this the label for the checkbox? If so, it needs to be a `label` element, and not a `div`. We also need to add `htmlFor` for accessibility.